smart_control/dataset/dataset.py

```diff
@@ -1,96 +1,405 @@
+        |  0 | 202194278473007104 | PHRED       | SB1:AHU:AC-2      |             6 | {'building_air_static_pressure_sensor': 1,  ... }     | {'exhaust_air_damper_percentage_command': 1, ... }
+        |  1 |   2760348383893915 | CDM         | VAV CO 1-1-10 CO2 |             4 | {'zone_air_heating_temperature_setpoint': 1, ... }    | {'supply_air_damper_percentage_command': 1, ... }                                                                                                                                                                                                    |
+        |  2 |   2562701969438717 | CDM         | VAV CO 2-2-36     |             4 | {'zone_air_heating_temperature_setpoint': 1, ... }    | {'supply_air_flowrate_setpoint': 1, ... }                                                                                                                                                                                                                                              |
+        |  3 |   2806035809406684 | CDM         |                   |             4 | {'discharge_air_temperature_setpoint': 1, ... }       | {'discharge_air_temperature_setpoint': 1, ... }                                                                                                                                                        |
+        |  4 |   2790439929052995 | CDM         | VAV CO 1-1-43     |             4 | {'zone_air_heating_temperature_setpoint': 1, ... }    | {'supply_air_damper_percentage_command': 1, ... }                                                                                                                                                                                                                                              |
     """
-    if building not in self.partitions:
-      raise ValueError("Invalid building")
-
-    if partition not in self.partitions[building]:
-      raise ValueError("invalid partition")
-
-    path = f"./{building}/tabular/{building}/{partition}/"
-
-    data = np.load(path + "data.npy.npz")
-    metadata = pickle.load(open(path + "metadata.pickle", "rb"))
-
-    if "device_infos" not in metadata.keys():
-      metadata["device_infos"] = pickle.load(
-          open(f"./{building}/tabular/device_info_dicts.pickle", "rb")
-      )
-    if "zone_infos" not in metadata.keys():
-      metadata["zone_infos"] = pickle.load(
-          open(f"./{building}/tabular/zone_info_dicts.pickle", "rb")
-      )
-    return data, metadata
+    # pylint: enable=line-too-long
+    df = pd.DataFrame(self.device_infos)
+    df = df.drop(columns=["zone_id"])  # many to many relationship
+    # consider renaming zone_id to zone_ids instead of dropping
+    return df
+
+  # DEVICE FIELDS
+
+  def _count_device_fields(self, field_type: str) -> pd.Series:
```

Comment:
If we value efficiency and avoiding the nested loop, we could consider using a [`polars`](https://pola.rs/) Dataframe and/or SQL Context to write a query to produce the same results. However if we value keeping the package dependencies to a minimum, we can move forward with the current approach, and circle back to improving efficiency in the future as needed.